corgidrp/detector.py

```diff
@@ -520,7 +525,178 @@ def calc_sat_fwc(emgain_arr,fwcpp_arr,fwcem_arr,sat_thresh):
+        bool_iniszero = np.logical_and(nrr<rad,full_residuals[ymin:ymax,xmin:xmax]==0)
+        bool_outisnotzero = np.logical_and(nrr>=rad,full_residuals[ymin:ymax,xmin:xmax]!=0)
+        
+        full_residuals[ymin:ymax,xmin:xmax][bool_innotzero] = (nrr_copy[bool_innotzero] + full_residuals[ymin:ymax,xmin:xmax][bool_innotzero])/2
+        full_residuals[ymin:ymax,xmin:xmax][bool_iniszero] = nrr_copy[bool_iniszero]
+        full_residuals[ymin:ymax,xmin:xmax][bool_outisnotzero] = full_residuals[ymin:ymax,xmin:xmax][bool_outisnotzero]
+        
+        
+        
+        full_qe_resel = ndimage.convolve(full_qe, mask)
+        full_residuals_resel = ndimage.convolve(full_residuals,mask)
+        std_resel = (((full_qe_resel - full_residuals_resel)/full_qe_resel))
+    
+    full_residuals[full_residuals==0] = None
+    nx = np.arange(0,std_resel.shape[1])
+    ny = np.arange(0,std_resel.shape[0])
+    nxx,nyy = np.meshgrid(ny,nx)
+    nrr = np.sqrt((nxx-n/2)**2 + (nyy-n/2)**2)
+    full_residuals[nrr>n_pix]= 1
+    std_resel[nrr>n_pix] = 0
+    full_qe[nrr>n_pix]= 1
+    full_qe=np.pad(full_qe, ((n_pad,n_pad),(n_pad,n_pad)), mode='constant',constant_values=(1))
+    full_residuals=np.pad(full_residuals, ((n_pad,n_pad),(n_pad,n_pad)), mode='constant',constant_values=(1))
+    std_resel=np.pad(std_resel, ((n_pad,n_pad),(n_pad,n_pad)), mode='constant',constant_values=(0))
+    
+    return (full_qe,full_residuals,std_resel)
+    
+    
+def create_onsky_flatfield(dataset, planet=None,band=None,up_radius=55,im_size=420,N=3,rad_mask=None,  planet_rad=None, n_pix=165, n_pad=302):
+    """Turn this dataset of raster scanned image frames of neptune or uranus into on-sky calibrated flat field 
+    
+        Args:
+            dataset (corgidrp.data.Dataset): a dataset of image frames that are raster scanned (L2a-level)
+            planet (str): neptune or uranus
+            band (str): 1 or 4
+            up_radius (int): Number of pixels on either side of centroided planet images (=55 pixels for Neptune and uranus)
+            im_size (int): x-dimension of the planet image (in pixels= 420 for the HST images)
+            N (int): Number of images to be combined for creating a matched filter (defaults to 3, can be changed)
+            rad_mask (float): radius in pixels used for creating a mask for band (band1=1.25, band4=1.75)
+            planet_rad (int): radius of the planet in pixels (planet_rad=50 for neptune, planet_rad=65)
+            n_pix (int): Number of pixels in radius covering the Roman CGI imaging FOV defaults to 165 pixels
+            n_pad (int): Number of pixels padded with '1s'  to generate the image size 1024X1024 pixels around imaging FOV (defaults to 302 pixels)
+
+
+    	Returns:
+    		data.FlatField (corgidrp.data.FlatField): a master flat for flat calibration using on sky images of planet in band specified
+    		
+	"""
+    if planet==None:
```

Comment:
general best practice is to use `is None` and not `== None` for none-checking. Although this isn't a huge deal, but should be a quick change.

Reply:
you use this in several places in this module

Reply:
Fixed it